src/Symfony/Component/Lock/Store/StoreFactory.php

```diff
@@ -27,6 +29,9 @@ class StoreFactory
     public static function createStore(#[\SensitiveParameter] object|string $connection): PersistingStoreInterface
     {
         switch (true) {
+            case $connection instanceof DynamoDbClient:
+                return new DynamoDbStore($connection);
```

Comment:
It would be nice to add a specific error message when the package is missing. I suggest adding a private function to this class.

```php
private static function requireBridgeClass(string $class, string $package): void
{
    if (!class_exists($class)) {
        throw new \LogicException(sprintf('Class "%s" is missing. Try running "composer require %s" to install the bridge.), $class, $package);
}
```

And then:
```suggestion
                $this->requireBridgeClass(DynamoDbStore::class, 'symfony/amazon-dynamodb-lock');

                return new DynamoDbStore($connection);
```

Reply:
Done